src/message_dao_test.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+import subprocess
+import zlib
+
+from unittest import mock
+from google.cloud.sql.connector import Connector
+
+import message
+import message_dao
+import archive_converter
+from test_helpers import test_data_path
+
+class MessageDaoTest(unittest.TestCase):
+
+    def setUp(self):
+        self.addCleanup(mock.patch.stopall)
+        self.mock_connect = mock.patch.object(Connector, 'connect').start()
+        self.mock_con = self.mock_connect.return_value
+        self.mock_cur = self.mock_con.cursor.return_value.__enter__.return_value
+        self.dao = message_dao.MessageDao('FAKE_GIT_PATH')
+
+    def assert_call_counts(self, connect, execute, commit):
+        self.assertEqual(connect, self.mock_connect.call_count)
+        self.assertEqual(execute, self.mock_cur.execute.call_count)
+        self.assertEqual(commit, self.mock_con.commit.call_count)
+
+    def test_init(self):
+        # Testing initialization of DAO
+        self.assert_call_counts(1, 3, 1)
```

Comment:
optional: wdyt about
```python
self.assert_call_counts(connect=1, excecute=3, commit=1)
```

I think this would make the tests more readable.

I still think it's a bit tricky to follow the tests, since mocks and call counts always are.
If we don't want calls to be made, we should probably make them throw errors by default.

E.g. I'd prefer we have

```python
def setUp(self):
  ...
  self.assert_call_counts(1, 3, 1)
  
  self.mock_con.reset_mock() # reset the calls() tracking for all the mocks
  # we can maybe also set self.mock_connect.side_effect = RuntimeError("should not be called again after init!")
  ...

def test_store(self):
  ...
  self.mock_cur.assert_called_once_with(..., mock.ANY)  # maybe match the
  self.mock_commit.assert_called_once()
  
  # then call reset_mock() again
```

wdyt?

Reply:
I like the idea of explicit keyword argument to make the test more readable. Although, I'm not sure if we'll need it if we're instead throwing errors for mocks that shouldn't be run; mock_connect only runs on init and mock_commit only runs when storing. Since mock_execute is the only one that consistently shows up, it's probably better to just have a single assertEquals(execute_count, mock_execute)

Reply:
Sorry, I should have made it more clear.

The suggestions are on a spectrum:
* one end: annotate the parameters to make it a bit easier to follow
* other end: move away from doing asserts on call counts and switch to explicit `.assert_called_once_with(...)`. Rely on errors to enforce unexpected calls aren't made.

We can do a mix of both.

E.g. we could make it so execute() is called once and then would error out if called again.
Then we'd do `assert_call_counts(execute=1, commit=1)`, etc.

I'd personally prefer we go farther to "throw errors" end.

Reply:
Is there a benefit in checking the value of a mocked cursor (aka `self.mock_cur.assert_called_once_with(..., mock.ANY)`)? Wouldn't it just be copying the code under the `store` method?

Reply:
I agree with this. It makes it much clearer.